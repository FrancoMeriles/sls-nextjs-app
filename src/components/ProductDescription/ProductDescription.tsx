import { Divider, Text } from '@chakra-ui/react'

const ProductDescription = ({ productDescription }) => {
  return (
    <>
      <Divider orientation="horizontal" />
      <Text fontSize="2xl" m="20px 0">
        Descripci√≥n
      </Text>
      <Text fontSize="lg" color="#666" lineHeight="35px">
        {productDescription.plain_text}
      </Text>
    </>
  )
}

export default ProductDescription
