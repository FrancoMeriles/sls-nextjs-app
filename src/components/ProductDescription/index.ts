import ProductDescription from './ProductDescription'

export default ProductDescription
