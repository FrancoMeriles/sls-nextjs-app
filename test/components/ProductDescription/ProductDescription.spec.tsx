import { render, screen } from '@testing-library/react'
import renderer from 'react-test-renderer'

import ProductDescription from '@base/components/ProductDescription'

const productDescription = {
  plain_text: '__LOREM_IPSUM_DOLOR__',
}
describe('ProductDescription', () => {
  it('should render and match snapshot with required props', () => {
    const tree = renderer
      .create(<ProductDescription productDescription={productDescription} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should renders and display the correct data', async () => {
    render(<ProductDescription productDescription={productDescription} />)
    expect(screen.getByText('Descripci√≥n')).toBeInTheDocument()
    expect(screen.getByText(productDescription.plain_text)).toBeInTheDocument()
  })
})
