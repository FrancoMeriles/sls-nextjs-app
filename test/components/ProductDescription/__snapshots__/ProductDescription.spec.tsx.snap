// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductDescription should render and match snapshot with required props 1`] = `
Array [
  <hr
    aria-orientation="horizontal"
    className="chakra-divider css-1upb9tn"
  />,
  <p
    className="chakra-text css-1hlj7v5"
  >
    Descripci√≥n
  </p>,
  <p
    className="chakra-text css-1hqh23w"
  >
    __LOREM_IPSUM_DOLOR__
  </p>,
]
`;
